lib/common/daemon.go

```diff
@@ -137,6 +138,22 @@ func ParseStringSlice(v *viper.Viper, key string) []string {
 	return vals
 }
 
+// MarshalConfig marshals a *viper.Viper config to JSON. pretty controls if the
+// result is indented or not. It replaces the masked fields with three
+// asterisks, if they are present.
+func MarshalConfig(v *viper.Viper, pretty bool, maskedFields ...string) ([]byte, error) {
+	all := v.AllSettings()
+	for _, f := range maskedFields {
+		if _, exists := all[f]; exists {
+			all[f] = "***"
+		}
```

Comment:
These lines remind me of the weirdness we deal with some weeks ago reg slices of strings in Viper.
Don't recall if we should have special consideration. I think not.